contracts/stark-proxy/v2/impl/SP2Exchange.sol

```diff
@@ -0,0 +1,285 @@
+        'SP2Exchange: Cannot deposit borrowed funds to the exchange while Restricted'
+      );
+    }
+
+    emit DepositedToExchange(starkKey, assetType, vaultId, tokenAmount);
+    return tokenAmount;
+  }
+
+  /**
+   * @notice Trigger a withdrawal of account funds held in the exchange contract. This can be
+   *  called after a (slow) withdrawal has already been processed by the L2 exchange.
+   *
+   * @param  starkKey   The STARK key of the account. Must be authorized by OWNER_ROLE.
+   * @param  assetType  The exchange asset ID for the asset to withdraw.
+   *
+   * @return The ERC20 token amount received by this contract.
+   */
+  function withdrawFromExchange(
+    uint256 starkKey,
+    uint256 assetType
+  )
+    external
+    nonReentrant
+    onlyRole(EXCHANGE_OPERATOR_ROLE)
+    onlyAllowedKey(starkKey)
+    returns (uint256)
+  {
+    return _withdrawFromExchange(starkKey, assetType, false);
+  }
+
+  // ============ Internal Functions ============
+
+  function _withdrawFromExchange(
+    uint256 starkKey,
+    uint256 assetType,
+    bool isGuardianAction
+  )
+    internal
+    returns (uint256)
+  {
+    uint256 startingBalance = getTokenBalance();
+    STARK_PERPETUAL.withdraw(starkKey, assetType);
+    uint256 endingBalance = getTokenBalance();
+    uint256 tokenAmount = endingBalance.sub(startingBalance);
+    emit WithdrewFromExchange(starkKey, assetType, tokenAmount, isGuardianAction);
+    return tokenAmount;
+  }
+
+  function _forcedWithdrawalRequest(
+    uint256 starkKey,
+    uint256 vaultId,
+    uint256 quantizedAmount,
+    bool premiumCost,
+    bool isGuardianAction
+  )
+    internal
+  {
+    STARK_PERPETUAL.forcedWithdrawalRequest(starkKey, vaultId, quantizedAmount, premiumCost);
+    emit RequestedForcedWithdrawal(starkKey, vaultId, isGuardianAction);
+  }
+
+  function _forcedTradeRequest(
+    uint256[12] calldata args,
+    bytes calldata signature,
+    bool isGuardianAction
+  )
+    internal
+  {
+    // Split into two functions to avoid error 'call stack too deep'.
+    if (args[11] != 0) {
+      _forcedTradeRequestPremiumCostTrue(args, signature);
+    } else {
+      _forcedTradeRequestPremiumCostFalse(args, signature);
+    }
+    emit RequestedForcedTrade(
+      args[0], // starkKeyA
+      args[2], // vaultIdA
+      isGuardianAction
+    );
+  }
+
+  function _depositCancel(
+    uint256 starkKey,
+    uint256 assetType,
+    uint256 vaultId,
+    bool isGuardianAction
+  )
+    internal 
+  {
+    STARK_PERPETUAL.depositCancel(starkKey, assetType, vaultId);
+    emit DepositCanceled(starkKey, assetType, vaultId, isGuardianAction);
+  }
+
+  function _depositReclaim(
+    uint256 starkKey,
+    uint256 assetType,
+    uint256 vaultId,
+    bool isGuardianAction
+  )
+    internal 
+  {
+    uint256 balanceBefore = TOKEN.balanceOf(address(this));
+    STARK_PERPETUAL.depositReclaim(starkKey, assetType, vaultId);
+    uint256 balanceAfter = TOKEN.balanceOf(address(this));
```

Comment:
For consistency/readability/maintainability could you match the style used above?

```
    uint256 startingBalance = getTokenBalance();
    ...
    uint256 endingBalance = getTokenBalance();
    uint256 tokenAmount = endingBalance.sub(startingBalance);
```